src/allmydata/immutable/upload.py

```diff
@@ -242,31 +242,26 @@ def get_uri_extension_data(self):
     def get_verifycapstr(self):
         return self._verifycapstr
 
-# our current uri_extension is 846 bytes for small files, a few bytes
-# more for larger ones (since the filesize is encoded in decimal in a
-# few places). Ask for a little bit more just in case we need it. If
-# the extension changes size, we can change EXTENSION_SIZE to
-# allocate a more accurate amount of space.
-EXTENSION_SIZE = 1000
```

Comment:
There is another constant representing this size defined at `allmydata.immutable.layout.ReadBucketProxy.MAX_UEB_SIZE`.  I don't _think_ it is strictly necessary for its value to exactly match the real UEB extension size but it probably is necessary that it be no smaller.  I'm not sure what the best way to deal with this is.  Maybe just a unit test?

Reply:
The actual size hasn't changed, so this branch doesn't make things worse... :grin: But I'll see if there's a reasonable way to do this.

Reply:
Actually, I should get rid of MAX_UEB_SIZE too, since part of the point of the exercise is to get rid of all out-of-bounds reads.